File: problems/1817-calculate-money-in-leetcode-bank/calculate-money-in-leetcode-bank.php
class Solution {

    /**
     * @param Integer $n
     * @return Integer
     */
    function totalMoney($n) {
        $result = 0;
        $step = 1;
        $initial = 1;
        
        for ($i = 1; $i <= $n; $i++) {
            $result += $initial;
            ++$initial;
            
            if ($i % 7 == 0) {
                ++$step;
                $initial = $step; 
            }
        }
        
        return $result;
    }
}

File: problems/1817-calculate-money-in-leetcode-bank/calculate-money-in-leetcode-bank.py
class Solution:
    def totalMoney(self, n: int) -> int:
        pre = 0
        cur = 0
        sum = 0
        for i in range(n):
            if i % 7 == 0:
                pre += 1
                cur = pre
            sum += cur
            cur += 1
        return sum